### PR TITLE
(WIP) Release 1.0.0

### LINKED ISSUE #184: Release 1.0.0
Larger changes are ahead. The download counter of the predecessor jfryman/selinux shows 4,203,495 downloads.

To clearly indicate by version numbers that whe introduce breaking changes, the next release should be labeld as 1.0.0 and shall be the last Puppet 3 module. 

The 1.0.0 will then be branched to the puppet3 branch and only accept bugfixes and added deprecation messages. 

Version 2.0.0 will contain Puppet 4 data types, new types/providers and streamlined parameter names in defined types. Most of the changes are breaking changes.

I've added milestones. You can view issues and PR's by milestone tag now:

Planned:
https://github.com/voxpupuli/puppet-selinux/milestones/1.0.0
https://github.com/voxpupuli/puppet-selinux/milestones/2.0.0

Unplanned:
https://github.com/voxpupuli/puppet-selinux/milestones/1.x.y
https://github.com/voxpupuli/puppet-selinux/milestones/2.x.y

### PR DESCRIPTION
**Pull request by vinzent**
Successfully ran acceptance test for beaker sets:

* centos-72-x64
* centos-66-x64
* fedora-25-x64
* fedora-24-x64
    
Additionally successfully tested with centos/6 and centos/7
distribution upstream vagrant boxes and the same beaker test.
    
Closes #184 

Included commits since 0.8.0 release not directly in this PR:

510c303 (GH-148) Add spec tests for class parameters
de69bf8 Use correct type in create_resource with permissive parameter
b9041d2 (GH-148) Make use of class selinux parameters
a25d44d (GH-119) Don't accept udp6 and tcp6 as protocol name with selinux::port
5744329 Fix for Security/YAMLLoad rubocop warning
211de87 Adapt required rubocop config change
0467ea8 modulesync 0.19.0
0bf10cc Fix broken link to puppet strings documentation
699e6b2 Generated puppet strings docs
0f83a6f Provide resource ordering UML diagram
d12a836 Convert inline docs to puppet-strings format
c0f1168 Add redcarpet gem needed for puppet strings
65ecdde Generate puppet strings in docs/ folder
7fd977a Document known problems / limitations
c34f99c [blacksmith] Bump version to 0.8.1-rc0
3442761 modulesync 0.18.0
a304af6 bump required puppet version to 4.6.1
4f99a3d (GH-147) Test default resource ordering in acceptance test
b306f49 (GH-147) Add spec tests for the ordering of resources
8116a4f (GH-147) Add ordering of resources

### CODE REVIEW

**Comment by vinzent** [comment]
rebased to latests master

**Comment by oranenj** [comment]
Seems fine to me at least. Let's get this out of the way and start breaking things. 😛

**Comment by bastelfreak** [comment]
Do we want to support puppet3 with this release? The CHANGELOG.md for the last release says that that one was the last with puppet3 support.

**Comment by bastelfreak** [comment]
we dropped all the puppet3 stuff on master with the last modulesync. if 1.0.0 should support puppet3 we have to sync again IMO.

---

I would be happy if 1.0.0 supports only puppet4

**Comment by rnelson0** [comment]
I read the changelog as saying it will start to introduce some actually-breaking changes soon but it hasn't yet, even though it's puppet 4 only from a testing standpoint.

**Comment by vinzent** [comment]
yes 1.0.0 should support puppet3, breaking changes will be in 2.0.0 as discussed in #184 .

but it seems the CHANGELOG needs clarification.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
 # Change Log
 
+## 2017-01-18 - Release 1.0.0
+
+This release marks the interface of this module as stable.
+
+The 1.x series will only accept bugfixes and added deprecation notices.
```
Comment (rnelson0) [inline CHANGELOG.md]: Strike from here to L12

**Comment by vinzent** [comment]
will not be release as 1.0.0. will be some 0.9.0 release.

**Pull request closed without merging**